app/src/main/java/com/devshish/internship/presentation/ui/library/SongViewHolder.kt

```diff
@@ -19,16 +20,18 @@ class SongViewHolder(
 
             tvSongName.text = song.title
             tvArtist.text = song.artist
-
-            val minutes = song.duration / 60
-            val seconds = song.duration % 60
-            val duration = String.format("%02d:%02d", minutes, seconds)
-            tvDuration.text = duration
+            tvDuration.text = String.format(
+                    "%02d:%02d",
+                    TimeUnit.MILLISECONDS.toMinutes(song.duration),
+                    TimeUnit.MILLISECONDS.toSeconds(song.duration) - TimeUnit.MINUTES.toSeconds(
+                        TimeUnit.MILLISECONDS.toMinutes(song.duration)
+                    )
+                )
```

Comment:
Something like this will be better: 
```
val minutes = MILLISECONDS.toMinutes(song.duration)
val seconds = MILLISECONDS.toSeconds(song.duration) - MINUTES.toSeconds(minutes)

tvDuration.text = String.format("%02d:%02d", minutes, seconds)
```

You can import TimeUnit by pressing alt-enter